gap/grahom.gi

```diff
@@ -522,9 +549,43 @@ function(src, ran, x)
     and IsInjectiveListTrans(DigraphVertices(src), x);
 end);
 
+InstallMethod(IsDigraphMonomorphism,
+"for digraph, digraph, transformation, list, list",
+[IsDigraph, IsDigraph, IsTransformation, IsList, IsList],
+function(src, ran, x, c1, c2)
+  return IsDigraphHomomorphism(src, ran, x, c1, c2)
+    and IsInjectiveListTrans(DigraphVertices(src), x);
+end);
+
 InstallMethod(IsDigraphMonomorphism, "for digraph, digraph, and perm",
 [IsDigraph, IsDigraph, IsPerm], IsDigraphHomomorphism);
 
+InstallMethod(IsDigraphMonomorphism, "for digraph, digraph, perm, list, list",
+[IsDigraph, IsDigraph, IsPerm, IsList, IsList], IsDigraphHomomorphism);
+
+InstallMethod(IsDigraphEmbedding,
+"for digraph, digraph by out-neighbours, transformation, list, and list",
+[IsDigraph, IsDigraphByOutNeighboursRep, IsTransformation, IsList, IsList],
+function(src, ran, x, c1, c2)
+  local y, induced, i, j;
+  if not IsDigraphMonomorphism(src, ran, x, c1, c2) then
```

Comment:
Is there some reason to not make the method for `IsDigraphEmbedding` analogous to the ones for `IsDigraphMonomorphism`? Or perhaps write a single function `RespectsColouring` so that all of these methods become: 


~~~~ 

function(src, ran, x, c1, c2)
return IsDigraphHomomorphism(src, ran, x) and RespectsColouring(src, ran, x);
end;
~~~~

where `IsDigraphHomomorphism` is replaced with `IsDigraphMonomorphism` or `IsDigraphEmbedding` as appropriate?